### PR TITLE
feat: run webpack build inside CI and add simple workflow for PRs

### PR DESCRIPTION
**Pull request by DRoet**
This removes the `dist/` folder from source control, the new github action will install all the dependencies and run the build for us.

This way we will never forget to run `run npm build` before publishing and have somewhat of a guarantee that the build is created successfully.

Having the CI/CD available also makes it easier to add automated tests in the future

@artf do you agree with this change? I could spin up a similar PR for the main grapesjs repo if you are a fan of this workflow

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "lint": "eslint src",
     "lint:fix": "eslint src --fix",
     "v:patch": "npm version --no-git-tag-version patch",
-    "build": "npm run v:patch && webpack -p",
+    "build": "webpack -p",
```
Comment (artf) [inline package.json:10]: without `npm version`, who will handle to create the tag and update the package.json version?

**Comment by artf** [comment]
not a big fan being dependent on platforms (in this case GH actions) but let's try for now with this repo

File: package.json (hunk 1 above)
Comment (DRoet) [reply package.json:10]: If you want to publish a new release, you just have to run `npm version patch` or `npm run v:patch` and push the new commit + tags.

Having the version increment every time you try and build locally is kind of annoying imo.

**Comment by DRoet** [comment]
> not a big fan being dependent on platforms (in this case GH actions) but let's try for now with this repo

You could still run `npm publish` locally but you just have to remember to build to project locally.

File: package.json (hunk 1 above)
Comment (artf) [reply package.json:10]: I mean, with the new workflow, if you need to publish the package on npm someone needs to update the version manually or it's handled by GH actions?

Comment (DRoet) [reply package.json:10]: Right now the steps are:

1. `npm run v:patch` to bump the version in package.json/package-lock.json and create a git tag
2. push the commit + new tag
3. go to github and draft a new release
4. select the new tag and add some release notes
5. hit publish                                                     `// after hitting this the "npm-publish.yml" GH action will publish to npm`

Does that clear things up? We could modify the workflow to skip steps 3-5 and let the GH action release as soon as the tags are pushed and add the release notes after.

Comment (artf) [reply package.json:10]: Ok, now it's clear. What about putting these steps in README? :)

Comment (DRoet) [reply package.json:10]: Sure I'll add a note somewhere

**Pull request merged**